### ISSUE TITLE
headless service update does not re-update listeners

### ISSUE THREAD

**Issue by howardjohn** [istio/istio#26617]
After scaling a headless service, push logs look like:

```
2020-08-18T22:18:07.052723Z     info    ads     Push debounce stable[4] 1: 100.034787ms since last change, 100.034345ms since last push, full=true
2020-08-18T22:18:07.054045Z     info    ads     XDS: Pushing:2020-08-18T15:18:07-07:00/3 Services:7 ConnectedEndpoints:1
```

No LDS update here. config_dump shows no LDS update as well. Restart istiod results in proper LDS

**Comment by howardjohn** [comment]
This is broken in 1.6 and master

**Comment by hzxuzhonghu** [comment]
Have you tested it on 1.7?

**Comment by howardjohn** [comment]
I think I only tried master

**Comment by hzxuzhonghu** [comment]
OK, let me test it

**Comment by hzxuzhonghu** [comment]
Reprocuded on 1.7 and i guess the cause is https://github.com/istio/istio/blob/cea39bd56c982699bff071b77a9f54ba7f42b66f/pilot/pkg/serviceregistry/kube/controller/endpointcontroller.go#L72

Here we should pass svc's hostname

**Comment by hzxuzhonghu** [comment]
fixed by #26636
